packages/loader/container-loader/src/containerContext.ts

```diff
@@ -279,7 +284,10 @@ export class ContainerContext implements IContainerContext {
         }
 
         for (const comparer of comparers) {
-            const satisfies = await comparer.satisfies(this.codeDetails, constraintCodeDetails);
+            const satisfies = await comparer.satisfies(
+                moduleWithDetails.details ?? this.codeDetails,
```

Comment:
this.codeDetails should be moduleWithDetails.details if it is a module with details

Reply:
Not sure, I understand. I do want to use module's details when provided and it's expressed in the code. When not provided the behavior will remain the same as earlier.

Reply:
Will revisit in a follow-up pr.